DuckDuckGo/NetworkProtectionTunnelController.swift

```diff
@@ -311,6 +317,31 @@ final class NetworkProtectionTunnelController: TunnelController {
         tunnelManager.onDemandRules = [NEOnDemandRuleConnect()]
     }
 
+    // MARK: - Observing Configuration Changes
+
+    private func subscribeToConfigurationChanges() {
+        notificationCenter.publisher(for: .NEVPNConfigurationChange)
+            .receive(on: DispatchQueue.main)
+            .sink { _ in
+                Task { @MainActor in
+                    guard let manager = await self.internalManager else {
+                        return
+                    }
+
+                    do {
+                        try await manager.loadFromPreferences()
+
+                        if manager.connection.status == .invalid {
+                            self.clearInternalManager()
+                        }
```

Comment:
This is the key to making sure that the shared manager is always valid. This publisher will clear it out, and the next time it's accessed then a new one will be created (assuming we have a configuration at all).

Reply:
Looking good - exactly as in macOS so I'm hoping that means it won't cause trouble.